### ISSUE TITLE
[DocDB] [DB Clone][YSQL] Create index fails in cloned DB

### ISSUE THREAD

**Issue by Arjun-yb** [yugabyte/yugabyte-db#21624]
Jira Link: [DB-10519](https://yugabyte.atlassian.net/browse/DB-10519)
### Description

**Version**: **2.23.0.0-b36**
**Steps**:

1. Create database(demo1) and create snapshot schedule
2. Create table(table1) and load some data
3. Collect time(t1) 
4. Create an index
5. Clone demo1 to demo2 at time t1
6. Connect to Cloned DB(demo2) and create an index to the table(table1)

**Observation**:
Create index fails with
`ERROR:  Index 0000400d000030008000000000004005 not found in index_map. Current schema is 6`
Describe table shows index as INVALID
```
Indexes:
    "t1_pkey" PRIMARY KEY, lsm (id HASH)
    "idx_4" lsm (name HASH) INVALID
```
Next create index works fine

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-10519]: https://yugabyte.atlassian.net/browse/DB-10519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by yamen-haddad** [comment]
This issue has the same root cause as https://github.com/yugabyte/yugabyte-db/issues/21016 and is fixed [here](https://github.com/yugabyte/yugabyte-db/commit/bb7c25d080d8883b5753710a975bc804bbc1492b). Tried to repro locally and the clone operation was successful.
I also managed to create an index on the target database and verified that we are able to use index scan in some select queries.
I am leaving the ticket open as I am planning to write a unit test to cover this case.
